bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -133,20 +136,12 @@ private boolean isTargetBranchAllowed() {
         if (jbsIssue.isEmpty()) {
             return Optional.empty();
         }
-        org.openjdk.skara.issuetracker.Issue csr = null;
-        for (var link : jbsIssue.get().links()) {
-            var relationship = link.relationship();
-            if (relationship.isEmpty() || !relationship.get().equals("csr for")) {
-                continue;
-            }
-            csr = link.issue().orElse(null);
-            if (csr == null) {
-                log.warning("The CSR " + link + " of the issue " + issue + " does not exist");
-            } else {
-                break;
-            }
-        }
-        if (csr != null) {
+        org.openjdk.skara.issuetracker.Issue csr = jbsIssue.get().links().stream()
+                .filter(link -> link.relationship().isPresent() && "csr for".equals(link.relationship().get()))
+                .findAny().flatMap(Link::issue).orElse(null);
+        if (csr == null) {
```

Comment:
I just noticed this pattern being used a lot. Instead of .orElse(null), I would prefer if you let csr be an optional and used `csr.isPresent()` in the if statement. You will then need to use `csr.get()` to actually get the value, but this is safe as it's protected by the condition on isPresent(). If the variable in question is used a lot, you can choose resolve it with .get() into a new variable at the top of the if/else block.

This applies to several places where the same pattern is used.

Reply:
Fixed.